rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -188,6 +188,7 @@ public static ScriptableObject initSafeStandardObjects(
 
         NativeArrayIterator.init(scope, sealed);
         NativeStringIterator.init(scope, sealed);
+        getRegExpProxy(cx).register(scope, sealed);
```

Comment:
I think that by doing this, we're changing the implementation so that it'll bomb if "RegExpImpl" isn't in the classpath, because "getRegExpProxy" will return null in that case. I'm not sure that we have ever taken advantage of that fact, but I think that with the current lazy constructor, a less catastrophic failure will result, and like all things Rhino, we don't really know what people are doing with the general capability to have the reg exp implementation be optional. Is there a different way to do this?

Reply:
Changed that in https://github.com/mozilla/rhino/pull/1731/commits/314de9e3321b8f8c1a4c28312dc5f15c19bee845